### PR TITLE
HV-1121 Fix property path generation for type argument constraints

### PR DESCRIPTION
**Pull request by gsmet**
* https://hibernate.atlassian.net/browse/HV-1121

We will have to revisit it later once BVAL-508 is included but I think it's a good first step.

### CODE REVIEW

File: engine/src/main/java/org/hibernate/validator/internal/engine/ValueContext.java [hunk 1]
```diff
@@ -211,6 +216,11 @@ public String toString() {
 		return sb.toString();
 	}
 
+	@Override
+	protected Object clone() {
```
Comment (gunnarmorling) [inline engine/src/main/java/org/hibernate/validator/internal/engine/ValueContext.java]: When do we clone it?

File: engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java [hunk 2]
```diff
@@ -772,11 +757,31 @@ else if ( isIndexable ) {
 		}
 	}
 
+	private ValueContext<?, Object> buildNewLocalExecutionContext(ValueContext<?, Object> valueContext, Object value) {
```
Comment (gunnarmorling) [inline engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java]: No actual change here besides the method extraction, right?

File: engine/src/main/java/org/hibernate/validator/internal/engine/ValueContext.java (hunk 1 above)
Comment (gsmet) [inline engine/src/main/java/org/hibernate/validator/internal/engine/ValueContext.java]: Oops, we don't. It's a leftover of some investigation.

File: engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java (hunk 2 above)
Comment (gsmet) [inline engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java]: Yes, no change but I needed to use the exact same code a few lines above so I extracted it.

**Comment by gsmet** [comment]
Merged and backported to 5.3.

**Pull request closed without merging**